From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: zfcp: Remove message for failed port
References: bnc#464466

Symptom:     During opening of an adapter the message "Remote port ...
             could not be opened" is emitted for initiator ports,
             confusing users.
Problem:     The port scan tries to open all ports, including
             initiator ports to determine if they are target ports.
             Sometimes, a different error status is returned for the
             initiator ports, triggering the message mentioned above.
Solution:    Remove the message, target port failures will be checked
             later in the error recovery, printing a different message
             if necessary.

Acked-by: John Jolly <jjolly@suse.de>
---
 Documentation/kmsg/s390/zfcp |   15 ---------------
 drivers/s390/scsi/zfcp_dbf.c |    2 +-
 drivers/s390/scsi/zfcp_fsf.c |    6 ------
 3 files changed, 1 insertion(+), 22 deletions(-)

--- a/Documentation/kmsg/s390/zfcp	2008-12-19 13:18:45.000000000 +0100
+++ b/Documentation/kmsg/s390/zfcp	2008-12-19 13:18:59.000000000 +0100
@@ -677,21 +677,6 @@
  */
 
 /*?
- * Text: "%s: Remote port 0x%016Lx could not be opened\n"
- * Severity: Warning
- * Parameter:
- *   @1: bus ID of the zfcp device
- *   @2: WWPN
- * Description:
- * The FCP adapter rejected a request to open the specified port. No retry
- * is possible.
- * User action:
- * Verify the setup and try removing and adding the port again. If this
- * problem persists, gather Linux debug data, collect the FCP adapter
- * hardware logs, and report the problem to your support organization.
- */
-
-/*?
  * Text: "%s: LUN 0x%Lx on port 0x%Lx is already in use by CSS%d, MIF Image ID %x\n"
  * Severity: Warning
  * Parameter:
--- a/drivers/s390/scsi/zfcp_dbf.c	2008-12-19 13:18:45.000000000 +0100
+++ b/drivers/s390/scsi/zfcp_dbf.c	2008-12-19 13:18:59.000000000 +0100
@@ -521,7 +521,7 @@ static const char *zfcp_rec_dbf_ids[] = 
 	[29]	= "link down",
 	[30]	= "link up status read",
 	[31]	= "open port failed",
-	[32]	= "open port failed",
+	[32]	= "",
 	[33]	= "close port",
 	[34]	= "open unit failed",
 	[35]	= "exclusive open unit failed",
--- a/drivers/s390/scsi/zfcp_fsf.c	2008-12-19 13:18:45.000000000 +0100
+++ b/drivers/s390/scsi/zfcp_fsf.c	2008-12-19 13:18:59.000000000 +0100
@@ -1405,13 +1405,7 @@ static void zfcp_fsf_open_port_handler(s
 		switch (header->fsf_status_qual.word[0]) {
 		case FSF_SQ_INVOKE_LINK_TEST_PROCEDURE:
 		case FSF_SQ_ULP_DEPENDENT_ERP_REQUIRED:
-			req->status |= ZFCP_STATUS_FSFREQ_ERROR;
-			break;
 		case FSF_SQ_NO_RETRY_POSSIBLE:
-			dev_warn(&req->adapter->ccw_device->dev,
-				 "Remote port 0x%016Lx could not be opened\n",
-				 (unsigned long long)port->wwpn);
-			zfcp_erp_port_failed(port, 32, req);
 			req->status |= ZFCP_STATUS_FSFREQ_ERROR;
 			break;
 		}
